common-test-lib/src/main/java/com/google/cloud/tools/intellij/testing/DelayedSubmitExecutorServiceProxy.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.testing;
+
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A proxy {@link ExecutorService} that delays the execution of {@link
+ * ExecutorService#submit(Runnable)}. This proxy instead stores the supplied {@link Runnable} and
+ * then executes it only after {@link #doSubmit()} is invoked.
+ */
+public class DelayedSubmitExecutorServiceProxy extends AbstractExecutorService {
+
+  private final ExecutorService executor;
+  private Runnable task;
+
+  /** Creates a new instance that is a proxy for the given {@code executor}. */
+  public DelayedSubmitExecutorServiceProxy(ExecutorService executor) {
+    this.executor = executor;
```

Comment:
Could you just initialize this within this class instead of passing it in? Since it's a test util, I would imagine every caller is going to want a direct executor service anyway.

Reply:
do feel strongly about this? Its not impossible to conceive of test usages of other types of executors.

Reply:
If a test wants to use anything other than a direct executor, then this wrapper becomes almost useless; there's not much value in determining when a task would execute on a different thread because you would still have to continuously check for completion from your test's thread, even when using this wrapper.

I'll leave this up to you.